Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2019,14 +2060,27 @@ static bool isContextMenuEvent(const WebMouseEvent& event)
             resnapAfterLayout();
         }
         break;
-    case ContextMenuItemTag::NextPage:
-        if (auto currentPageIndex = indexForCurrentPageInView(); currentPageIndex < m_documentLayout.pageCount() - 1)
-            scrollToPage(currentPageIndex.value() + 1);
+    case ContextMenuItemTag::NextPage: {
+        auto currentPageIndex = indexForCurrentPageInView();
+        auto pageCount = m_documentLayout.pageCount();
+        auto pagesPerRow = m_documentLayout.pagesPerRow();
+
+        if (currentPageIndex < pageCount - 1 && (m_documentLayout.isSinglePageDisplayMode() || (m_documentLayout.isTwoUpDisplayMode() && currentPageIndex % pagesPerRow)))
```

Comment:
Was a bit hard to follow this condition, maybe if we could self-document the condition with a function name and then call it? (I don't have a suggestion since I'm not 100% sure what the condition is)

Reply:
```cpp
if (canMoveToNextPage(...)) {
    pageIncrement = ...
    scrollToPage(currentPageIndex + pageIncrement);
}
```

Reply:
Yeah agreed that this is a lot to parse. Added a lambda and some helpers so hopefully this is a bit better